### PR TITLE
feat(extract): always extract files into its own folder

### LINKED ISSUE #11537: extract Plugin: Can automatically create the decompression directory
### If the feature request is for a plugin or theme, specify it here.

extract plugin

### If the feature solves a problem you have, specify it here.

If there is no package folder in the compressed file, a bunch of scattered files may appear after using extract plugin to extract.

### Describe the proposed feature.

For compressed files without folder package inside, can they be automatically put into a folder after decompression

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by mcornella**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [ ] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

- Creates a folder based on file name and extracts the contents from within this folder
- Remove the folder if there was an error
- If the only content in the extraction folder is one folder, move the folder up and remove the created extraction folder

Fixes #11537

### CODE REVIEW

File: plugins/extract/extract.plugin.zsh [hunk 1]
```diff
@@ -29,61 +29,75 @@ EOF
...
 
-    # Go back to original working directory in case we ran cd previously
+    # Go back to original working directory
+    # and remove extraction directory if there was an error
     builtin cd -q "$pwd"
+    (( success > 0 )) && command rm -r "$extract_dir"
+
+    # If content of extract dir is a single directory, move its contents up
+    # Glob flags: N: no error if directory is empty, Y2: at most 2 files
+    local content=("${extract_dir}"/*(NY2))
```
Comment (carlosala) [inline plugins/extract/extract.plugin.zsh:97]: Being a local array, shouldn't be declared and then initialized, or that is just with associative arrays?

Comment (carlosala) [inline plugins/extract/extract.plugin.zsh:97]: Here `$content` does not take hidden files into account I'd say.

**Review by carlosala — COMMENTED** [review]
Generally looks good to me! Just those two small things, really nice improvement Marc! 😊

File: plugins/extract/extract.plugin.zsh (hunk 1 above)
Comment (mcornella) [reply plugins/extract/extract.plugin.zsh:97]: Added `D` ("glob dots") glob flag to account for that as well.

**Review by carlosala — APPROVED** [review]
Super fast fix! I learnt a new glob qualifier haha :)
LGTM😉

**Pull request merged**